pkg/daemon/update.go

```diff
@@ -2179,7 +2179,7 @@ func (dn *CoreOSDaemon) applyLayeredOSChanges(mcDiff machineConfigDiff, oldConfi
 
 	var osExtensionsContentDir string
 	var err error
-	if mcDiff.extensions || mcDiff.kernelType {
+	if mcDiff.osUpdate || mcDiff.extensions || mcDiff.kernelType {
```

Comment:
I wouldn't block on this, but I think it would also be nice to avoid pulling the extensions if we don't have any layered packages in the new config.

Basically keep the "happy path" optimal of "node update == pull and boot a container".